### PR TITLE
fix dead channels giveaway

### PR DESCRIPTION
**Pull request by dziaineka**
Fox gives to application the pid of the rabbit channel so that the application can do its business through it. Sometimes the application kills the channel process.

For example, when queue parameters have changed in config and the application creates this queue on top of an existing queue with the old parameters. In this case the process dies and we take a new one and create the queue again, killing the old one.

The dead process pid remains in the channel process pool and will be given to the application again when the queue comes to it.

I bring a fix for fox to check if the channel pid is dead before giving it out.

### CODE REVIEW

File: src/publish/fox_pub_pool.erl [hunk 1]
```diff
@@ -160,3 +153,30 @@ terminate(_Reason, _State) ->
...
+
+    {Channel1, Channels2} = case NumChannels < PoolSize of
+        true ->
+            {ok, Channel} = amqp_connection:open_channel(Connection),
+            {Channel, Channels};
+        false ->
+            {{value, Channel}, Channels1} = queue:out(Channels),
+            {Channel, Channels1}
+    end,
+
```
Comment (yzh44yzh) [inline src/publish/fox_pub_pool.erl:172]: It is not a big deal, however we can check if process alive only when we get it from queue. No need to check newly created channel.

Comment (dziaineka) [reply src/publish/fox_pub_pool.erl:172]: Yeah. Thank you. 
This place don't seems to be heavy loaded by channels creation so my intention was to make code flatter, better readable.

Place "👀" as a reaction to the message to force me fixing it 🙃

Comment (yzh44yzh) [reply src/publish/fox_pub_pool.erl:172]: No need, it's not a big deal )

**Comment by yzh44yzh** [comment]
@dziaineka tag 2.4.3

**Pull request merged**